crates/trigger/src/runtime_config.rs

```diff
@@ -182,6 +184,58 @@ impl RuntimeConfig {
         }
     }
 
+    // returns the client tls options in form of nested
+    // HashMap of { Component ID -> HashMap of { Host -> ParsedClientTlsOpts} }
+    pub fn client_tls_opts(
+        &self,
+    ) -> Result<HashMap<String, HashMap<Authority, ParsedClientTlsOpts>>> {
+        let mut components_map: HashMap<String, HashMap<Authority, ParsedClientTlsOpts>> =
+            HashMap::new();
+
+        // if available, use the existing client tls opts value for a given component-id and host-authority
+        // to ensure first-one wins incase of duplicate options
+        fn use_existing_if_available(
+            existing_opts: Option<&HashMap<Authority, ParsedClientTlsOpts>>,
+            host: Authority,
+            newopts: ParsedClientTlsOpts,
+        ) -> Option<(Authority, ParsedClientTlsOpts)> {
```

Comment:
It looks like this function never returns `None`. Can this just return `(Authority, ParsedClientTlsOpts)` instead of the option?